e2e-tests/src/cases.rs

```diff
@@ -0,0 +1,32 @@
+use std::collections::HashMap;
+
+use crate::test::{
+    batch_transactions as test_batch_transactions, change_validators as test_change_validators,
+    channeling_fee as test_channeling_fee, fee_calculation as test_fee_calculation,
+    finalization as test_finalization, staking_era_payouts as test_staking_era_payouts,
+    staking_new_validator as test_staking_new_validator, token_transfer as test_token_transfer,
+    treasury_access as test_treasury_access,
+};
+use crate::config::Config;
+
+/// Wrapper around the type of the test case.
+#[derive(Copy, Clone)]
+pub enum TestCase {
+    BlockNumberResult(fn(&Config) -> anyhow::Result<u32>),
```

Comment:
we dont really use the return value from only one tests that returns something. I would rather rewrite the function `test_finalization` so it returns unit and remove the TestCase enum.

Reply:
In hindsight, this indeed seems like a more sensible choice, changed.